Data/PuzzleServerContext.cs

```diff
@@ -76,6 +77,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             modelBuilder.Entity<SinglePlayerPuzzleStatePerPlayer>().HasIndex(pspt => new { pspt.PlayerID });
             modelBuilder.Entity<SinglePlayerPuzzleStatePerPlayer>().HasIndex(pspt => new { pspt.PlayerID, pspt.SolvedTime });
             modelBuilder.Entity<SinglePlayerPuzzleUnlockState>().HasKey(state => new { state.PuzzleID });
+            modelBuilder.Entity<Room>().HasIndex(room => new { room.EventID, room.Building, room.Number }).IsUnique();
```

Comment:
Need to check for delete issues by ensuring deleting teams doesn't delete rooms and deleting events works

Reply:
Deleting teams now sets the teamid to null. Deleting events deletes the rooms, which is intended unless someone tells me otherwise.